src/instance/InstanceController.ts

```diff
@@ -1428,4 +1430,51 @@ export class ServerController {
       res.status(createInternalServerError(error).code).json(createInternalServerError(error));
     }
   }
+
+  @Get('/log')
+  @ApiOperation({summary: 'Get the complete log as plain text.'})
+  @ApiResponse({
+    schema: {
+      example: 'string',
+    },
+    status: 200,
+  })
+  @ApiResponse(status500description)
+  async getLog(@Res() res: Response): Promise<void> {
+    try {
+      let logData = '';
+
+      if (errorLogFile) {
+        logData += `=== ${errorLogFile} ===\n`;
+        try {
+          const errorLogData = await fs.readFile(errorLogFile, {encoding: 'utf8'});
+          logData += errorLogData;
+        } catch (error) {
+          logger.error(error);
+          logData += `Error: Could not find error log file "${errorLogFile}" or it is not readable.`;
+        }
+      } else {
+        logData += `Error: No error log file specified.`;
+      }
+
+      logData += '\n';
+
+      if (outLogFile) {
+        logData += `=== ${outLogFile} ===\n`;
+        try {
+          const outLogData = await fs.readFile(outLogFile, {encoding: 'utf8'});
+          logData += outLogData;
+        } catch (error) {
+          logger.error(error);
+          logData += `Error: Could not find output log file "${outLogFile}" or it is not readable.`;
+        }
+      } else {
+        logData += `Error: No output log file specified.`;
+      }
+
+      res.contentType('text/plain; charset=UTF-8').send(logData);
```

Comment:
Can we please have `logData` as string[] and do a logData.join('\n') here? That way you won't need to add '\n' everywhere manually

Reply:
Good one..